src/systemTest/java/com/flightstats/hub/system/functional/ChannelMaxItemsEnforcementTest.java

```diff
@@ -0,0 +1,106 @@
+package com.flightstats.hub.system.functional;
+
+import com.flightstats.hub.model.ChannelConfig;
+import com.flightstats.hub.model.ChannelType;
+import com.flightstats.hub.system.extension.TestClassWrapper;
+import com.flightstats.hub.system.service.ChannelConfigService;
+import com.flightstats.hub.system.service.ChannelItemCreator;
+import com.flightstats.hub.system.service.ChannelItemRetriever;
+import com.flightstats.hub.system.service.HubInternalService;
+import com.flightstats.hub.system.service.S3Service;
+import lombok.SneakyThrows;
+import org.awaitility.Awaitility;
+import org.awaitility.Duration;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+import javax.inject.Inject;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static com.flightstats.hub.util.StringUtils.randomAlphaNumeric;
+
+@Execution(ExecutionMode.SAME_THREAD)
+class ChannelMaxItemsEnforcementTest extends TestClassWrapper {
+    private static final String TEST_DATA = "TEST_DATA";
+    private final List<String> items = new ArrayList<>();
+    private String channelName;
+    @Inject
+    private ChannelConfigService channelConfigService;
+    @Inject
+    private ChannelItemCreator channelItemCreator;
+    @Inject
+    private ChannelItemRetriever channelItemRetriever;
+    @Inject
+    private HubInternalService hubInternalService;
+    @Inject
+    private S3Service s3Service;
+
+    @AfterEach
+    void cleanup() {
+        items.clear();
+        channelConfigService.delete(channelName);
+    }
+
+    @SneakyThrows
+    private void addItemsOverTime() {
+        /*
+            batch items are compared with granularity of one minute
+            so if the excess items are of the same minute they will not be deleted
+        */
+        items.add(channelItemCreator.addItem(channelName, TEST_DATA));
+        Thread.sleep(TimeUnit.MINUTES.toMillis(1));
+        items.addAll(channelItemCreator.addItems(channelName, TEST_DATA, 5));
+    }
+
+    private void waitUntilAllItemsAreInS3(ChannelType type) {
+        Awaitility.await().pollInterval(Duration.FIVE_SECONDS)
+                .atMost(new Duration(3, TimeUnit.MINUTES))
+                .until(() -> s3Service.confirmItemsInS3(type, items, channelName) &&
+                        items.stream().allMatch(path -> channelItemRetriever.getItem(path).isPresent()));
+    }
+
+    private String setupTest(ChannelType type) {
```

Comment:
`createChannelAndAddTooManyItems` might be a better name for this method. I think that might indicate this method is hiding too much stuff though. I'd break it into something like a `createChannelWithMaxItems(5)`, `addItem` (which saves the path), `wait`, `addItems(5)` step, especially since there's only one test method in this class.